### PR TITLE
BUG: Fix OpenAI compatibility issue during chat

### LINKED ISSUE #1113: BUG: OpenAI Stream API 实现未完全按照OpenAI 格式输出，可能导致部分OpenAI API Client 实现调用报错
### Describe the bug

使用流式请求时 ，在[OpenAI 的API 官方文档](https://platform.openai.com/docs/api-reference/chat/create)中，SSE的第一个事件， content 字段的值是空字符串，而在inference 的OpenAI API 实现中， SSE的第一个事件，content 字段字段不存在， 相当于  content = null，部分OpenAI API Client 实现中，对content字段做了NotNull 校验(如Spring AI)，这将会导致客户端调用异常

inference 响应
```
data: {"id": "chata240e594-ded3-11ee-bd37-1a757c6c50cc", "model": "chatglm3", "created": 1710071103, "object": "chat.completion.chunk", "choices": [{"index": 0, "delta": {"role": "assistant"}, "finish_reason": null}]}

data: {"id": "chata240e594-ded3-11ee-bd37-1a757c6c50cc", "model": "chatglm3", "created": 1710071103, "object": "chat.completion.chunk", "choices": [{"index": 0, "delta": {"content": ""}, "finish_reason": null}]}
```

OpenAI 响应
```
{"id":"chatcmpl-123","object":"chat.completion.chunk","created":1694268190,"model":"gpt-3.5-turbo-0125", "system_fingerprint": "fp_44709d6fcb", "choices":[{"index":0,"delta":{"role":"assistant","content":""},"logprobs":null,"finish_reason":null}]}

{"id":"chatcmpl-123","object":"chat.completion.chunk","created":1694268190,"model":"gpt-3.5-turbo-0125", "system_fingerprint": "fp_44709d6fcb", "choices":[{"index":0,"delta":{"content":"Hello"},"logprobs":null,"finish_reason":null}]}
```

查看相关代码实现后，第一条消息固定返回如图：
image

只需要在此处添加一行·"content"：”“` , 即可

### To Reproduce

inference version 0.9.2

### Expected behavior

与OpenAI 官方API 输出保持一致 

### Additional context

稍后我会尝试提交PR修复

### LINKED ISSUE #1143: OpenAI Compatible API Incompatibility
### Describe the bug
I am reporting an issue encountered while using the xinference library version 0.9.1 with the qwen1.5 model. When attempting to leverage the OpenAI-compatible API, specifically during the invocation of the `ConversableAgent._reflection_with_llm()` function in Autogen, , which automatically appends a summary system message at the end of a dialogue, I encounter a BadRequestError with the following message:

`BadRequestError: Error code: 400 - {'detail': 'System message should be the first one.'}`

This error arises due to the system message not being the first message in the message history. However, this issue does not occur when using an official OpenAI key.

### To Reproduce

1. Deploy the xinference: 0.9.1 Docker container with the qwen1.5-chat model.
2. Full error stack trace: `BadRequestError: Error code: 400 - {'detail': 'System message should be the first one.'}`
3. Simplified reproduction code snippet using curl:
```
curl http://localhost:9997/v1/chat/completions   -H "Content-Type: application/json"   -H "Authorization: Bearer sk-anykey"   -d '{
    "model": "qwen1.5",
    "messages": [
      {
        "role": "user",
        "content": "What is triangle inequality?"
      },
      {
        "role": "assistant",
        "content":"The triangle inequality is a fundamental geometric concept that states that the sum of the lengths of any two sides of a triangle is always greater than or equal to the length of the third side. In other words, in a triangle, if you have two sides a and b, and a third side c, the following inequality must hold:\n\na + b > c\nb + c > a\nc + a > b\n\nThis inequality ensures that the three sides create a valid geometric triangle, as it prevents the formation of a \"triangle\" where the third side is shorter than the combined length of the other two. It is a simple yet essential principle in geometry, particularly in measuring distances and understanding the properties of triangles."
      },
      {
        "role": "system",
        "content": "Summarize the takeaway from the conversation. Do not add any introductory phrases."
      }
    ]
  }'
```

### Expected behavior
xinference's openai compatible API should handle message histories consistently with the actual OpenAI API, even when system messages are not at the beginning of the message sequence.

### Additional context
It would be greatly appreciated if the developers could look into improving the compatibility of xinference's openai compatible API to better align with the official OpenAI API behavior.

Thank you for your attention to this matter!

### LINKED ISSUE #1158: BUG: Multiple system messages are not supported.
### To Reproduce
To help us to reproduce this bug, please provide information below:

1. Python 3.11
2. xinference 0.9.3
3. Versions of crucial packages.
4. Full stack of the error.
5. Minimized code to reproduce the error.

### Expected behavior
When running `xinference-local --log-level debug`, use the following command to launch the qwen model:
```
xinference launch --model-name qwen1.5-chat --size-in-billions 1_8 --model-format pytorch -u my-llm
```
Then attempting to integrate complex prompt instructions using the Agently framework, the output results in an error.
```python
#test.py
import Agently

agent_factory = (
    Agently.AgentFactory()
        .set_settings("model.OpenAI.auth.api_key", "nothing")
        .set_settings("model.OpenAI.options", {"model": "my-llm"})
        .set_settings("model.OpenAI.url", "http://127.0.0.1:9997/v1")
)

agent = agent_factory.create_agent()
result = (
    agent
        # general: agent应该知道的全局信息，通常被视作system prompt或类似位置的信息
        .general("输出规定", "必须使用中文进行输出")
        # role: agent自身的角色设定信息
        .role({
            "姓名": "大脑",
            "任务": "作大脑，根据场景判断用户需求，调用相关功能满足用户需求",
        })
        # user_info: agent需要了解的用户相关的信息
        .user_info("给出输入信息的是一个可以判断当前场景和用户意图的程序")
        # abstract: 对于之前对话（尤其是较长对话）的总结信息
        .abstract(None)
        # chat_history: 按照OpenAI消息列格式的对话记录list
        ## 支持:
        ## [{ "role": "system", "content": "" },
        ##  { "role": "assistant", "content": "" },
        ##  { "role": "user", "content": "" }]
        ## 三种角色
        .chat_history([])
        # input: 和本次请求相关的输入信息
        .input({
            "question": "当前场景XX；用户意图：xx，需要舒缓情绪",
            "reply_style_expect": "{commend: [], reason:[]}"
        })
        # info: 为本次请求提供的额外补充信息
        .info("车")
        # instruct: 为本次请求提供的行动指导信息
        .instruct([
            "请用，按照{reply_style_expect}风格输出",
        ])
        # output: 对本次请求的输出提出格式和内容的要求
        .output({
            "reply": ("str", "按照{reply_style_expect}风格输出"),
            "next_questions": ([
                ("str",
                 "根据{reply}内容，结合{user_info}提供的用户信息，" +
                 "给用户推荐的可以进一步提问的问题"
                )], "不少于3个"),
        })
        # start: 用于开始本次主要交互请求
        .start()
)

print(result)
```

The error message is as follows:
```
 $ python test.py
Exception in thread Thread-2 (start_in_theard):
Traceback (most recent call last):
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/Agently/Agent/Agent.py", line 219, in start_in_theard
    reply = loop.run_until_complete(self.start_async(request_type))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/Agently/Agent/Agent.py", line 207, in start_async
    raise(e)
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/Agently/Agent/Agent.py", line 160, in start_async
    event_generator = await self.request.get_event_generator(request_type)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/Agently/Request/Request.py", line 102, in get_event_generator
    response_generator = await request_plugin_export["request_model"](request_data)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/Agently/plugins/request/OpenAI.py", line 203, in request_model
    return await self.request_gpt(request_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/Agently/plugins/request/OpenAI.py", line 186, in request_gpt
    stream = await client.chat.completions.create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/openai/resources/chat/completions.py", line 1330, in create
    return await self._post(
           ^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/openai/_base_client.py", line 1725, in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/openai/_base_client.py", line 1428, in request
    return await self._request(
           ^^^^^^^^^^^^^^^^^^^^
  File "/fs/fast/u2021201733/envs/inf/lib/python3.11/site-packages/openai/_base_client.py", line 1519, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'detail': 'Multiple system messages are not supported.'}
None
```

### PR DESCRIPTION
**Pull request by mujin2**
Fixes #1143 
Fixes #1113
Fixes #1158

### CODE REVIEW

**Comment by mujin2** [comment]
fixes #1158

File: xinference/api/restful_api.py [hunk 1]
```diff
@@ -1258,25 +1259,21 @@ async def create_chat_completion(self, request: Request) -> Response:
                 status_code=400, detail="Invalid input. Please specify the prompt."
             )
 
-        system_messages = []
+        system_messages: List["ChatCompletionMessage"] = []
```
Comment (ChengjieLi28) [inline xinference/api/restful_api.py:1262]: The typing should be `List[str]`

File: xinference/api/restful_api.py [hunk 2]
```diff
@@ -1258,25 +1259,21 @@ async def create_chat_completion(self, request: Request) -> Response:
...
+        system_messages_tmp = []
```
Comment (ChengjieLi28) [inline xinference/api/restful_api.py:1263]: Avoid using `tmp` as variable name.
Suggestion: `system_message_contents`

File: xinference/api/restful_api.py [hunk 3]
```diff
@@ -1258,25 +1259,21 @@ async def create_chat_completion(self, request: Request) -> Response:
...
             assert (
                 msg.get("content") != SPECIAL_TOOL_PROMPT
             ), f"Invalid message content {SPECIAL_TOOL_PROMPT}"
             if msg["role"] == "system":
-                system_messages.append(msg)
+                system_messages_tmp.append(msg["content"])
             else:
                 non_system_messages.append(msg)
+        system_messages.insert(
+            0, {"role": "system", "content": ".".join(system_messages_tmp)}
```
Comment (ChengjieLi28) [inline xinference/api/restful_api.py:1274]: Add a space to the separator: `. `

File: xinference/api/restful_api.py [hunk 4]
```diff
@@ -1258,25 +1259,21 @@ async def create_chat_completion(self, request: Request) -> Response:
...
         for msg in messages:
             assert (
                 msg.get("content") != SPECIAL_TOOL_PROMPT
             ), f"Invalid message content {SPECIAL_TOOL_PROMPT}"
             if msg["role"] == "system":
-                system_messages.append(msg)
+                system_messages_tmp.append(msg["content"])
             else:
                 non_system_messages.append(msg)
+        system_messages.insert(
```
Comment (ChengjieLi28) [inline xinference/api/restful_api.py:1273]: Just use `append` since it's empty for `system_messages`.

**Review by ChengjieLi28 — APPROVED** [review]

**Pull request merged**